owid/catalog/datasets.py

```diff
@@ -123,20 +124,28 @@ def save(self) -> None:
             warnings.warn(f"Dataset {self.metadata.short_name} is missing namespace")
 
         self.metadata.save(self._index_file)
-        self._update_table_metadata()
 
-    def _update_table_metadata(self) -> None:
-        """Update the copy of this datasets metadata in every table in the set."""
-        dataset_meta = self.metadata.to_dict()
+        # Update the copy of this datasets metadata in every table in the set.
+        for table_name in self.table_names:
+            table = self[table_name]
+            table.metadata.dataset = self.metadata
+            table._save_metadata(join(self.path, table.metadata.checked_name + ".meta.json"))
 
-        for metadata_file in glob(join(self.path, "*.meta.json")):
-            with open(metadata_file) as istream:
-                table_meta = json.load(istream)
-
-            table_meta["dataset"] = dataset_meta
+    def update_metadata(self, metadata_path: Path) -> None:
+        """
+        Load YAML file with metadata from given path and update metadata of dataset and its tables.
 
-            with open(metadata_file, "w") as ostream:
-                json.dump(table_meta, ostream, indent=2, default=str)
+        :param metadata_path: Path to *.meta.yml file with metadata. Check out other metadata files
+            for examples, this function doesn't do schema validation
+        """
+        self.metadata.update_from_yaml(metadata_path, if_source_exists="replace")
```

Comment:
Shouldn't `if_source_exists` also be an argument of `update_metadata`?

Reply:
I'll add it. I wasn't sure whether anyone is using anything other than `replace` and didn't want to add complexity.